src/test/scala/chiselTests/Inout.scala

```diff
@@ -0,0 +1,132 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+import chisel3._
+import chisel3.util._
+import chisel3.testers.BasicTester
+
+// IO for Modules that just connect bus to out
+class InoutReaderIO extends Bundle {
+  val bus = Analog(32.W)
+  val out = Output(UInt(32.W))
+}
+// IO for Modules that drive bus from in (there should be only 1)
+class InoutWriterIO extends Bundle {
+  val bus = Analog(32.W)
+  val in = Input(UInt(32.W))
+}
+
+trait InoutReader {
+  self: Module =>
+  final val io = self.IO(new InoutReaderIO)
+}
+
+class InoutReaderBlackBox extends BlackBox with InoutReader
+
+class InoutReaderWrapper extends Module with InoutReader {
+  val mod = Module(new InoutReaderBlackBox)
+  io <> mod.io
+}
+class InoutWriterBlackBox extends BlackBox {
+  val io = IO(new InoutWriterIO)
+}
+// Connects two Analog ports
+class InoutConnector extends Module {
+  val io = IO(new Bundle {
+    val bus1 = Input(Analog(32.W))
+    val bus2 = Input(Analog(32.W))
+  })
+  io.bus1 <> io.bus2
+}
+
+// Parent class for tests connecing up InoutReaders and InoutWriters
+abstract class InoutTester extends BasicTester {
+  final val BusValue = "hdeadbeef".U
+
+  final val (cycle, done) = Counter(true.B, 2)
+  when (done) { stop(); stop() } // Double stop for Verilator
```

Comment:
Unless this double stop pertains to Analog in particular, this should be filed as an issue and get fixed at the root.